Metadata handling

matkrak opened the pull request:
Changes in both curve.py and profile.py that provide metadata handling

matkrak committed:
Update curve.py

Added metadata to Curve class, provided passing informations in change_domain function by using __dict__

matkrak committed:
Update profile.py

Provided inheritance of metadata to subclasses using super()

grzanka commented on beprof/curve.py:
```diff
@@ -31,8 +31,12 @@ class Curve(np.ndarray):
+        if meta is None:
+            obj.metadata = {}
+        else:
+            obj.metadata = meta.copy()
```
I see you used shallow copy. Do you think it is better here than deep copy (https://docs.python.org/3.5/library/copy.html) ?

matkrak replied:
Deep copy is what I wanted to do, my bad

grzanka commented on beprof/curve.py:
```diff
@@ -31,8 +31,12 @@ class Curve(np.ndarray):
 
-    def __new__(cls, input_array):
+    def __new__(cls, input_array, **meta):
         obj = np.asarray(input_array).view(cls).copy()
```
Copying on the view here is also a subject of discussion. It doesn't necessarily need to stay here

matkrak replied:
Works without .copy() but that was something you wrote so I didn't dare to remove it!

matkrak committed:
Update curve.py

1) Deep copy instead of copy (metadata)
2) Removed copy() from Curve.__new__

matkrak commented:
If you don't have any other comments, shall I rewrite tests as docstring and issue can be closed?

grzanka commented on beprof/curve.py:
```diff
@@ -68,10 +73,44 @@ def y_at_x(self, x):
+              "\nMetadata : " + str(self.metadata)
         return ret
 
+    # functions not ready yet, added for testing metadata purpose
```
Remove it from this PR. It doesn't have anything in common with metadata handling.

matkrak committed:
Removed testing functions

Since I know that changing the domain won't affect metadata, I removed functions that were not subject of this PR.

matkrak commented:
Should I add any docstring tests here? Didn't add any new method after all. I should add some text to docstring of Curve anyway - not a word about metadata for now.

grzanka commented:
Add some docstring to Curve class describing how does it work.

matkrak committed:
Updated docstring

Described metadata in docstring
